rfcs/xxxx-v5-content-api.md

```diff
@@ -0,0 +1,217 @@
+            "publishedAt": "2023-01-01T00:00:00.000Z"
+          }
+        }
+      }
+    }
+  },
+  "meta": {
+    "pagination": {
+      "page": 1,
+      "pageSize": 10
+    }
+  }
+}
+```
+
+**After**
+
+```json
+{
+  "data": {
+    "documentId": "clkgylmcc000008lcdd868feh",
+    "entryId": "cpmnyztbcc964008lcft812feo",
+    "title": "Article A",
+    "relation": {
+      "documentId": "clkgylw7d000108lc4rw1bb6s",
+      "entryId": 1,
+      "name": "Category A",
+      "meta": {
+        "locale": "en",
+        "createdAt": "2023-01-01T00:00:00.000Z",
+        "updatedAt": "2023-01-01T00:00:00.000Z",
+        "publishedAt": "2023-01-01T00:00:00.000Z"
+      }
+    },
+    "meta": {
+      "locale": "fr",
+      "createdAt": "2023-01-01T00:00:00.000Z",
+      "updatedAt": "2023-01-01T00:00:00.000Z",
+      "publishedAt": "2023-01-01T00:00:00.000Z"
+    }
+  },
+  "meta": {
+    "pagination": {
+      "page": 1,
+      "pageSize": 10
+    }
+  }
+}
+```
+
+### Separating meta attributes to avoid any current & future name conflicts
+
+```json
+{
+  "data": {
+    "documentId": "clkgylmcc000008lcdd868feh",
+    "title": "Article A",
+    "description": "My description",
+    "meta": {
+      "locale": "",
+      "publishedAt": "",
+      "createdAt": "",
+      "createdBy": ""
+    }
+  }
+}
+```
+
+Pros:
+
+- No conflicts between user & system attributes
+- Clear separation in different objects
+- Easier extraction of all meta fields in one go
+
+Cons
+
+- Access is nested.
+- Lesser discoverability (need to know it’s in meta) ⇒ the _SDK and/or Typescript typings would make the discoverability issue irrelevant_
+
+```tsx
+const locale = data.meta.locale;
+const { locale } = data.meta;
+
+// better separation
+const { meta, ...attributes } = data;
+```
+
+### Low level **relational support**
+
+In order to keep the old relational system, we will return the `entryId` to allow managing low level relations.
+
+```json
+{
+  "data": {
+    "id": "clkgylmcc000008lcdd868feh",
+    "entryId": 1,
+    "title": "..."
+  }
+}
+```
+
+# Release plan
+
+Doing theses changes would break all the existing API Calls.
+
+In order to reduce migration cost we are considering support a _deprecated_ response format:
+
+- Keep the "id" name for the entry id
+- Support fetching the API with the old IDs
+- Keep the nested `data.attributes` format of v4
+
+```tsx
+{
+  "data": {
+    "documentId": "clkgylmcc000008lcdd868feh"
+    "id": "clkgylmcc000008lcdd868feh",
+    "attributes": {},
+  }
+}
+```
+
+- This format could be configured to be the default one during migration.
+- We would support a query parameter to switch to the new format incrementally.
+
+> The breaking changes are still forcing users to switch from calling the api with `id` to `documentId`
+
+# Unresolved questions
+
+- Should we use `id` or `documentId` in the new response format.
+- Should we use `entryId` or `entityId` to represent the underlying Ids to use for low level relations
```

Comment:
The API should not rename fields, so if you want to use either of it, the database should be renamed from id to e?id as well.